### PR TITLE
#137 Making NSURLSession extensible

### PR DESCRIPTION
**Pull request by shahmharsh**
Implements #137

### CODE REVIEW

**Comment by shahmharsh** [comment]
@WilliamDenniss

**Comment by iainmcgin** [comment]
@shahmharsh I see the signed individual contributor CLA for you, thanks for doing that. @protocol86 will be reviewing this change for you.

File: UnitTests/OIDURLSessionProviderTests.m [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    // Setting the session back to default sharedSession for future test cases
+    [OIDURLSessionProvider setSession:[NSURLSession sharedSession]];
+}
+
+- (void)testCustomSession {
+    NSURLSessionConfiguration *config = [NSURLSessionConfiguration ephemeralSessionConfiguration];
+    NSURLSession *customSession = [NSURLSession sessionWithConfiguration:config];
+    [OIDURLSessionProvider setSession:customSession];
+    NSURLSession *session = [OIDURLSessionProvider session];
+    XCTAssertEqual(session, customSession);
```
Comment (protocol86) [inline UnitTests/OIDURLSessionProviderTests.m]: Please use `XCTAssertEqualObjects` here.

File: Source/OIDURLSessionProvider.m [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "OIDURLSessionProvider.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSURLSession * __nullable gURLSession;
```
Comment (protocol86) [inline Source/OIDURLSessionProvider.m]: Remove space after "*"

**Comment by shahmharsh** [comment]
@protocol86 comments addressed

**Comment by shahmharsh** [comment]
When can I expect this to be merged?

**Comment by shahmharsh** [comment]
@WilliamDenniss?

**Pull request merged**